File: AndroidApp/app/src/main/java/com/jamespfluger/alexadevicefinder/models/DeviceSettings.java
package com.jamespfluger.alexadevicefinder.models;

import com.google.gson.annotations.SerializedName;

public class DeviceSettings {

    @SerializedName("alexaUserId")
    private String alexaUserId;
    @SerializedName("deviceId")
    private String deviceId;
    @SerializedName("deviceName")
    private String deviceName;
    @SerializedName("useFlashlight")
    private boolean useFlashlight;
    @SerializedName("useVibrate")
    private boolean useVibrate;
    @SerializedName("shouldLimitToWifi")
    private boolean shouldLimitToWifi;
    @SerializedName("configuredWifiSsid")
    private String configuredWifiSsid;
    @SerializedName("useVolumeOverride")
    private boolean useVolumeOverride;
    @SerializedName("overriddenVolumeValue")
    private int overriddenVolumeValue;

    public DeviceSettings() {
    }

    public DeviceSettings(String alexaUserId, String deviceId) {
        this.alexaUserId = alexaUserId;
        this.deviceId = deviceId;
    }

    public String getAlexaUserId() {
        return alexaUserId;
    }

    public void setAlexaUserId(String alexaUserId) {
        this.alexaUserId = alexaUserId;
    }

    public String getDeviceId() {
        return deviceId;
    }

    public void setDeviceId(String deviceId) {
        this.deviceId = deviceId;
    }

    public String getDeviceName() {
        return deviceName;
    }

    public void setDeviceName(String deviceName) {
        this.deviceName = deviceName;
    }

    public boolean getUseFlashlight() {
        return useFlashlight;
    }

    public void setUseFlashlight(boolean useFlashlight) {
        this.useFlashlight = useFlashlight;
    }

    public boolean getUseVibrate() {
        return useVibrate;
    }

    public void setUseVibrate(boolean useVibrate) {
        this.useVibrate = useVibrate;
    }

    public boolean getShouldLimitToWifi() {
        return shouldLimitToWifi;
    }

    public void setShouldLimitToWifi(boolean shouldLimitToWifi) {
        this.shouldLimitToWifi = shouldLimitToWifi;
    }

    public String getConfiguredWifiSsid() {
        return configuredWifiSsid;
    }

    public void setConfiguredWifiSsid(String configuredWifiSsid) {
        this.configuredWifiSsid = configuredWifiSsid;
    }

    public boolean getUseVolumeOverride() {
        return useVolumeOverride;
    }

    public void setUseVolumeOverride(boolean useVolumeOverride) {
        this.useVolumeOverride = useVolumeOverride;
    }

    public int getOverriddenVolumeValue() {
        return overriddenVolumeValue;
    }

    public void setOverriddenVolumeValue(int overriddenVolumeValue) {
        this.overriddenVolumeValue = overriddenVolumeValue;
    }

}

File: SharedModels/SharedModels/Devices/DeviceSettings.cs
using System;
using Amazon.DynamoDBv2.DataModel;

namespace DeviceFinder.Models.Devices
{
    [DynamoDBTable("DeviceFinder_DeviceSettings")]
    public class DeviceSettings
    {
        [DynamoDBHashKey("AlexaUserID")]
        public string AlexaUserId { get; set; }

        [DynamoDBRangeKey("DeviceID")]
        public string DeviceId { get; set; }

        [DynamoDBProperty("DeviceName")]
        public string DeviceName { get; set; }

        [DynamoDBProperty("UseFlashlight")]
        public bool UseFlashlight { get; set; }

        [DynamoDBProperty("UseVibrate")]
        public bool UseVibrate { get; set; }

        [DynamoDBProperty("ShouldLimitToWifi")]
        public bool ShouldLimitToWifi { get; set; }

        [DynamoDBProperty("ConfiguredWifiSsid")]
        public string ConfiguredWifiSsid { get; set; }

        [DynamoDBProperty("UseVolumeOverride")]
        public bool UseVolumeOverride { get; set; }

        [DynamoDBProperty("OverriddenVolumeValue")]
        public int OverriddenVolumeValue { get; set; }

        [DynamoDBProperty("ModifiedDate")]
        public DateTime ModifiedDate { get; }

        public DeviceSettings()
        {
            ModifiedDate = DateTime.UtcNow;
        }
    }
}
